File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        
        dict = {}

        for i in range(len(nums)):

            diff = target - nums[i]

            if diff in dict :
                return [i, dict[diff]]

            dict[nums[i]] = i

        return []

File: 0001-two-sum/0001-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {

       HashMap<Integer,Integer> hMap = new HashMap<>();

       for(int i=0; i< nums.length; i++)
       {
            int complement = target - nums[i];

            if(hMap.containsKey(complement))
            {
                return new int[]{i,hMap.get(complement)};
            }
            hMap.put(nums[i], i);
       } 
       return new int[]{};
    }
}